app/src/test/java/com/github/bgabriel998/softwaredevproject/utils/AngleLowpassFilterTest.java

```diff
@@ -0,0 +1,57 @@
+package com.github.bgabriel998.softwaredevproject.utils;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class AngleLowpassFilterTest {
+
+    // Tests the average between 0 and PI/2
+    @Test
+    public void lowpassFilterTest() {
+        AngleLowpassFilter lowpassFilter = new AngleLowpassFilter();
+
+        lowpassFilter.add(0);
+        assertEquals(0, lowpassFilter.average(), 0.01);
+
+        lowpassFilter.add((float) Math.PI/2);
+        assertEquals(Math.PI/4, lowpassFilter.average(), 0.01);
+    }
+
+    //Tests that the oldest element of the queue gets removed
+    @Test
+    public void lowpassFilterFullTest() {
```

Comment:
If I understood correctly, in this test you add 6 values because the constant `LENGTH` in the filter is set to 5, so you can test that after you added `LENGHT + 1` angles the filter has removed one angle.
But what if we decide to increase the constant? I feel like we should modify this test using a for and the constant `LENGTH`, so it remains consistent even if we change `LENGTH`. Tell me what you think!

Reply:
Agreed, is it possible to make this a for/while loop to make it more maintainable?

Reply:
Yes, I agree, I just updated the test with values that are also better to understand. The test now increments the value every time by PI/4 so that the average can be simply calculated by hand.